pkg/parsers/operatingsystem/operatingsystem_windows.go

```diff
@@ -8,40 +8,53 @@ import (
 
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
+	os, err := withCurrentVersionRegistryKey(func(key registry.Key) (os string, err error) {
+		if os, _, err = key.GetStringValue("ProductName"); err != nil {
+			return "", err
+		}
 
-	// Default return value
-	ret := "Unknown Operating System"
+		releaseId, _, err := key.GetStringValue("ReleaseId")
+		if err != nil {
+			return
+		}
+		os = fmt.Sprintf("%s Version %s", os, releaseId)
 
-	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
-	if err != nil {
-		return ret, err
-	}
-	defer k.Close()
+		buildNumber, _, err := key.GetStringValue("CurrentBuildNumber")
+		if err != nil {
+			return
+		}
+		ubr, _, err := key.GetIntegerValue("UBR")
+		if err != nil {
+			return
+		}
+		os = fmt.Sprintf("%s (OS Build %s.%d)", os, buildNumber, ubr)
 
-	pn, _, err := k.GetStringValue("ProductName")
-	if err != nil {
-		return ret, err
-	}
-	ret = pn
+		return
+	})
 
-	ri, _, err := k.GetStringValue("ReleaseId")
-	if err != nil {
-		return ret, err
+	if os == "" {
+		// Default return value
+		os = "Unknown Operating System"
 	}
-	ret = fmt.Sprintf("%s Version %s", ret, ri)
 
-	cbn, _, err := k.GetStringValue("CurrentBuildNumber")
-	if err != nil {
-		return ret, err
-	}
+	return os, err
+}
 
-	ubr, _, err := k.GetIntegerValue("UBR")
+// GetOperatingSystemVersion gets the version of the current operating system, as a string.
+func GetOperatingSystemVersion() (string, error) {
+	return withCurrentVersionRegistryKey(func(key registry.Key) (string, error) {
+		releaseId, _, err := key.GetStringValue("ReleaseId")
+		return releaseId, err
+	})
+}
+
+func withCurrentVersionRegistryKey(f func(registry.Key) (string, error)) (string, error) {
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
```

Comment:
This one gives the major/minor/patch version of Windows (eg 6.2.9200); whereas as far as I can tell Windows folks tend to prefer using the release ID to refer to Windows Server versions (see eg https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/system-requirements#virtualized-container-hosts or any of our own doc around Windows.

But I might be totally wrong on this, and I'm happy to change to whichever Windows-savvy people think makes more sense; since this patch is aimed primarily at UCP folks anyway, maybe @wsong or @alexmavr can give some insight here as to which version number is relevant here? Or maybe @dhiltgen ?